src/features/session_trace/instrument/index.js

```diff
@@ -71,57 +58,23 @@ export class Instrument extends InstrumentBase {
       handle('bstHist', [location.pathname + location.hash, this.startPath, this.time], undefined, FEATURE_NAMES.sessionTrace, thisInstrumentEE)
     })
 
-    if (supportsPerformanceObserver()) {
-      // capture initial resources, in case our observer missed anything
-      handle(BST_RESOURCE, [window.performance.getEntriesByType('resource')], undefined, FEATURE_NAMES.sessionTrace, thisInstrumentEE)
-
-      observeResourceTimings()
-    } else {
-      // collect resource timings once when buffer is full
-      if (window.performance[CRT] && window.performance[ADD_EVENT_LISTENER]) {
-        window.performance.addEventListener(RESOURCE_TIMING_BUFFER_FULL, this.onResourceTimingBufferFull, eventListenerOpts(false))
-      }
-    }
-
-    function observeResourceTimings () {
-      var observer = new PerformanceObserver((list, observer) => { // eslint-disable-line no-undef
-        var entries = list.getEntries()
-
+    let observer
+    try {
+      // Capture initial resources and watch for future ones. Don't defer this given there's a default cap on the number of buffered entries.
+      observer = new PerformanceObserver((list) => { // eslint-disable-line no-undef
+        const entries = list.getEntries()
         handle(BST_RESOURCE, [entries], undefined, FEATURE_NAMES.sessionTrace, thisInstrumentEE)
       })
-
-      try {
-        observer.observe({ entryTypes: ['resource'] })
-      } catch (e) {
-        // do nothing
-      }
+      observer.observe({ type: RESOURCE, buffered: true })
+    } catch (e) {
+      // Per NEWRELIC-8525, we don't have a fallback for capturing resources for older versions that don't support PO at this time.
     }
 
-    document.addEventListener('scroll', this.noOp, eventListenerOpts(false))
-    document.addEventListener('keypress', this.noOp, eventListenerOpts(false))
-    document.addEventListener('click', this.noOp, eventListenerOpts(false))
-
-    this.abortHandler = this.#abort
-    this.importAggregator()
-  }
-
-  /** Restoration and resource release tasks to be done if Session trace loader is being aborted. Unwind changes to globals. */
-  #abort () {
-    window.performance.removeEventListener(RESOURCE_TIMING_BUFFER_FULL, this.onResourceTimingBufferFull, false)
-    // The doc interaction noOp listeners are harmless--cannot buffer data into EE.
-    this.abortHandler = undefined // weakly allow this abort op to run only once
-  }
-
-  noOp (e) { /* no-op */ }
+    // document.addEventListener('scroll', noOp, eventListenerOpts(false))
```

Comment:
what was the point of these?

Reply:
If these are not needed, can we remove the commented out code?

Reply:
Removed